File: hl-momentum/src/utils/telegram.py
import requests
from src.config import config
from src.utils.logger import get_logger

log = get_logger(__name__)

def send_alert(message: str) -> None:
    """Send a Telegram message. Silently skips if token/chat not configured."""
    if not config.TELEGRAM_BOT_TOKEN or not config.TELEGRAM_CHAT_ID:
        return
    try:
        url = f"https://api.telegram.org/bot{config.TELEGRAM_BOT_TOKEN}/sendMessage"
        requests.post(url, json={
            "chat_id":    config.TELEGRAM_CHAT_ID,
            "text":       message,
            "parse_mode": "Markdown",
        }, timeout=5)
    except Exception:
        log.warning("Telegram alert failed")

File: sol-scanner/src/utils/telegram.ts
import axios from "axios";
import { config } from "../config";
import { log } from "./logger";

export async function sendAlert(message: string): Promise<void> {
  if (!config.telegram.botToken || !config.telegram.chatId) return;

  try {
    await axios.post(
      `https://api.telegram.org/bot${config.telegram.botToken}/sendMessage`,
      { chat_id: config.telegram.chatId, text: message, parse_mode: "Markdown" }
    );
  } catch {
    log.warn("Telegram alert failed");
  }
}
